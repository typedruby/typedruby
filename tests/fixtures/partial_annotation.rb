def f1
end

def f2(Integer x)
end

def f3(Integer x, y, Integer z)
end

def f4(x) => Integer
end

def f5(x, y, Integer z) => Integer
end

def f6 => Integer
end

def f7(Integer x, Integer y) => Integer
end

def f8[U]
end
