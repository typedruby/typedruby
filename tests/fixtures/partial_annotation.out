info: Typechecking...
error: Partial type signatures are not permitted in method definitions

        @ __ROOT__/tests/fixtures/partial_annotation.rb:6
      6 |  def f2(Integer x)
                 ^^^^^^^^^^^ all arguments and return value must be annotated

error: Partial type signatures are not permitted in method definitions

        @ __ROOT__/tests/fixtures/partial_annotation.rb:9
      9 |  def f3(Integer x, y, Integer z)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^ all arguments and return value must be annotated

error: Partial type signatures are not permitted in method definitions

        @ __ROOT__/tests/fixtures/partial_annotation.rb:12
     12 |  def f4(x) => Integer
                 ^^^^^^^^^^^^^^ all arguments and return value must be annotated

error: Partial type signatures are not permitted in method definitions

        @ __ROOT__/tests/fixtures/partial_annotation.rb:15
     15 |  def f5(x, y, Integer z) => Integer
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ all arguments and return value must be annotated

error: Partial type signatures are not permitted in method definitions

        @ __ROOT__/tests/fixtures/partial_annotation.rb:24
     24 |  def f8[U]
                 ^^^ all arguments and return value must be annotated
